### PR TITLE
Add snow kernel config, refactor executables

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

https://github.com/aws/eks-anywhere/issues/4294

*Description of changes:*

- Add hardcoded bottlerocket kernel config for snow
- Make executables a package
- Add `mocks` make target for generating mock files 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/aws/bottlerocket-bootstrap/pkg/executables/apiclient.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	Executable
+}
+
+func NewAPIClient() *APIClient {
+	return &APIClient{
+		Executable: NewExecutable(utils.ApiclientBinary),
+	}
+}
+
+func (a *APIClient) SetKuberentesCloudProvider(cloudProvider string) error {
```
Comment (vivek-koppuru) [inline projects/aws/bottlerocket-bootstrap/pkg/executables/apiclient.go:20]: Mispelled for all
```suggestion
func (a *APIClient) SetKubernetesCloudProvider(cloudProvider string) error {
```

File: projects/aws/bottlerocket-bootstrap/pkg/executables/apiclient.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+func (a *APIClient) SetKuberentesAllowUnsafeSysctls(sysctls []string) error {
+	q := make([]string, len(sysctls))
+	for idx, ctl := range sysctls {
+		q[idx] = fmt.Sprintf("%q", ctl)
+	}
+	_, err := a.Execute("set", fmt.Sprintf("allowed-unsafe-sysctls=[%s]", strings.Join(q[:], ",")))
+	return err
+}
+
+func (a *APIClient) SetKernelRmenMax(size string) error {
```
Comment (vivek-koppuru) [inline projects/aws/bottlerocket-bootstrap/pkg/executables/apiclient.go:44]: and below
```suggestion
func (a *APIClient) SetKernelRmemMax(size string) error {
```

File: projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/kernel.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package system
+
+import "github.com/pkg/errors"
+
+const (
+	maxSocketBufferSize = "8388608"
```
Comment (vivek-koppuru) [inline projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/kernel.go:6]: comment where you get this from?

File: projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/kernel.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	kernelCorePattern   = "/var/corefile/core.%e.%p.%h.%t"
+)
+
+func (s *Snow) configureKernelSettings() error {
+	if err := s.api.SetKernelRmenMax(maxSocketBufferSize); err != nil {
+		return errors.Wrap(err, "Error setting kernel maximum socket receive buffer size")
```
Comment (vivek-koppuru) [inline projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/kernel.go:12]: this probably needs to change throughout tho
```suggestion
		return errors.Wrap(err, "setting kernel maximum socket receive buffer size")
```

Comment (jiayiwang7) [reply projects/aws/bottlerocket-bootstrap/pkg/providers/snow/system/kernel.go:12]: i will have a separate pr for this

File: projects/aws/bottlerocket-bootstrap/pkg/executables/apiclient_test.go [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+			if tc.wantError {
+				tt.Expect(err).NotTo(BeNil())
+			} else {
+				tt.Expect(err).To(BeNil())
+			}
+		})
+	}
+}
+
+func TestSetkubernetesProviderID(t *testing.T) {
```
Comment (vivek-koppuru) [inline projects/aws/bottlerocket-bootstrap/pkg/executables/apiclient_test.go:129]:
```suggestion
func TestSetKubernetesProviderID(t *testing.T) {
```

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**